packages/spark/sodasql/dialects/spark_dialect.py

```diff
@@ -57,6 +57,9 @@ def sql_tables_metadata_query(self, limit: Optional[int] = None, filter: str = N
 
     def sql_tables_metadata(self, limit: str = 10, filter: str = None):
         # TODO Implement limit
+        if self.database is None:
```

Comment:
I think we should fix this together with [this issue](https://github.com/sodadata/soda-sql/issues/485). If not database is given, then we first query for all schemas/database and then we return all tables within each of the schemas/databases here.